kernel/packages/shared/apis/PortableExperiences.ts

```diff
@@ -0,0 +1,85 @@
+import { registerAPI, exposeMethod } from 'decentraland-rpc/lib/host'
+import { spawnPortableExperienceScene, killPortableExperienceScene } from 'unity-interface/portableExperiencesUtils'
+import { ExposableAPI } from './ExposableAPI'
+import { ParcelIdentity } from './ParcelIdentity'
+
+enum ExecutorType {
+  SCENE = 'SCENE',
+  WEARABLE = 'WEARABLE',
+  QUEST_UI = 'QUEST_UI'
+}
+type Executor = {
+  type: ExecutorType
+  identifier: string
+}
+type PortableExperienceIdentifier = string
+type PortableExperienceHandle = {
+  pid: string
+  parentProcess: Executor
+}
+type SpawnPortableExperienceParameters = {
+  urn: PortableExperienceIdentifier
+}
+
+@registerAPI('PortableExperiences')
+export class PortableExperiences extends ExposableAPI {
+  private currentPortableExperiences: Map<string, Executor> = new Map()
```

Comment:
Imagine this scenario:
- You go to 1,1
- LAND1,1 spawns "PE1"
- In order to complete the quest, you go to LAND150,150
- For distance reasons, scene in LAND1,1 gets unloaded
- In the process of scene unloading the state of `LAND1,1.PortableExperience.currentPortableExperiences` is garbage collected and emptied.
- You go back to 1,1, LAND1,1 is loaded again.
- LAND1,1 wants to kill the PE1 (it should be able to, since LAND1,1 created PE1)

I believe that having this map here couples the implementation to an ephemeral state.

As a solution to the previous example, I suggest to store the information of who spawned the portable experience in the portable experience itself. Like this

```ts
const portableExperienceScene = await spawnPortableExperienceScene(spawnParams.urn)
portableExperienceScene.spawnerCID = this.options.getAPIInstance(ParcelIdentity).cid
```

That way, all the comparisons will have a unique source of truth and the data would be more normalized (in database normalization terms).


That way, the implementation is very much straight forward without JSON parsing
```ts
async kill(pid: string){
  const peToKill = getPortableExperienceBySceneId(pid)
  const selfCID = this.options.getAPIInstance(ParcelIdentity).cid
  if (peToKill.spawnerCID == selfCID) {
    // can kill
  } else {
    // can't
  }
}
```